rkt/fly.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build linux
+
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/code.google.com/p/go-uuid/uuid"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/spf13/cobra"
+	"github.com/coreos/rkt/pkg/aci"
+	"github.com/coreos/rkt/store"
+)
+
+var (
+	cmdFly = &cobra.Command{
+		Use:   "fly IMAGE [ -- image-args...]",
+		Short: "Run a single application image with no pod or isolation",
+		Long:  `IMAGE should be a string referencing an image; either a hash, local file on disk, or URL.`,
+		Run:   runWrapper(runFly),
+	}
+)
+
+func init() {
+	cmdRkt.AddCommand(cmdFly)
+
+	// Disable interspersed flags to stop parsing after the first non flag
+	// argument. All the subsequent parsing will be done by parseApps.
+	// This is needed to correctly handle image args
+	cmdFly.Flags().SetInterspersed(false)
+}
+
+func runFly(cmd *cobra.Command, args []string) (exit int) {
+	err := parseApps(&rktApps, args, cmd.Flags(), true)
```

Comment:
Almost certainly, I don't like the global var, was just cribbing from rkt
run

On Thu, Jun 18, 2015 at 3:31 PM, Eugene Yakubovich <notifications@github.com
> wrote:

> In rkt/fly.go
> <https://github.com/coreos/rkt/pull/1072#discussion_r32785941>:
>
> > +		Long:  `IMAGE should be a string referencing an image; either a hash, local file on disk, or URL.`,
> > +		Run:   runWrapper(runFly),
> > +	}
> > +)
> > +
> > +func init() {
> > +	cmdRkt.AddCommand(cmdFly)
> > +
> > +	// Disable interspersed flags to stop parsing after the first non flag
> > +	// argument. All the subsequent parsing will be done by parseApps.
> > +	// This is needed to correctly handle image args
> > +	cmdFly.Flags().SetInterspersed(false)
> > +}
> > +
> > +func runFly(cmd *cobra.Command, args []string) (exit int) {
> > +	err := parseApps(&rktApps, args, cmd.Flags(), true)
>
> where's rktApps coming from? if it's a global var somewhere, can it be
> refactored to make it local somehow?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/rkt/pull/1072/files#r32785941>.
>